caffe2/core/registry.h

```diff
@@ -152,7 +152,7 @@ class CAFFE2_API Registerer {
  */
 #define CAFFE_DECLARE_TYPED_REGISTRY(                                    \
     RegistryName, SrcType, ObjectType, PtrType, ...)                     \
-  CAFFE2_EXPORT Registry<SrcType, PtrType<ObjectType>, ##__VA_ARGS__>*   \
```

Comment:
Any reason for CAFFE2_EXPORT a declaration? It'll lead to undefined reference when using it outside the main proj.